skrub/_sentence_encoder.py

```diff
@@ -0,0 +1,300 @@
+
+        if not sbd.is_string(column):
+            raise RejectColumn(f"Column {sbd.name(column)!r} does not contain strings.")
+
+        self._check_params()
+
+        self.cache_folder_ = get_data_dir(
+            name=self.model_name, data_home=self.cache_folder
+        )
+        try:
+            self.estimator_ = st.SentenceTransformer(
+                self.model_name,
+                device=self.device,
+                cache_folder=self.cache_folder_,
+                use_auth_token=self.use_auth_token,
+            )
+        except OSError as e:
+            raise ModelNotFound(
+                f"{self.model_name} is not a local folder and is not a valid "
+                "model identifier listed on 'https://huggingface.co/models'.\n "
+                "If this is a private repository, make sure to pass a token having "
+                "permission to this repo by passing `use_auth_token=<your_token>`"
+            ) from e
+
+        self.input_name_ = sbd.name(column)
+
+        X_out = self._vectorize(column)
+
+        if self.n_components is not None:
+            if (min_shape := min(X_out.shape)) >= self.n_components:
+                self.pca_ = PCA(
+                    n_components=self.n_components,
+                    copy=False,
+                    random_state=self.random_state,
+                )
+                X_out = self.pca_.fit_transform(X_out)
+            else:
+                warnings.warn(
+                    f"The matrix shape is {(X_out.shape)}, and its minimum is "
+                    f"{min_shape}, which is too small to fit a PCA with "
+                    f"n_components={self.n_components}. "
+                    "The embeddings will be truncated by keeping the first "
+                    f"{self.n_components} dimensions instead. "
+                    "Set n_components=None to keep all dimensions and remove "
+                    "this warning."
+                )
+                X_out = X_out[:, : self.n_components]
+
+        self.n_components_ = X_out.shape[1]
+
+        cols = self.get_feature_names_out()
+        X_out = sbd.make_dataframe_like(column, dict(zip(cols, X_out.T)))
+        X_out = sbd.copy_index(column, X_out)
+
+        return X_out
+
+    def transform(self, column):
+        """Transform ``column`` using the SentenceEncoder.
+
+        This method uses the embedding model loaded in memory during ``fit``
+        or ``fit_transform``.
+
+        Parameters
+        ----------
+        column : pandas or polars Series of shape (n_samples,)
+            The string column to compute embeddings from.
+
+        Returns
+        -------
+        X_out : pandas or polars DataFrame of shape (n_samples, n_components)
+            The embedding representation of the input.
+        """
+        check_is_fitted(self, "estimator_")
+
+        if not sbd.is_string(column):
+            raise RejectColumn(f"Column {sbd.name(column)!r} does not contain strings.")
```

Comment:
I should have written clearer documentation about RejectColumn; it is only meant to be used during fit/fit_transform.

the reason is that it is a way to signal to the meta-estimator (like the tablevectorizer) "I'm not the right transformer for this column, pick someone else" .

but if we don't reject the column during fit, then assignments will not change so at transform time it is too late to change our mind: we have accepted the responsibility to handle this column and now we must deal with it. if the data passed during transform is not appropriate, we can raise an exception (like ValueError or anything else, but not RejectColumn because RejectColumn will receive no special treatment at this point), or output NaNs or apply whatever strategy seems most appropriate.

TLDR keep the RejectColumn in fit_transform, but in `transform`, either raise a different exception or output NaNs

Reply:
> the reason is that it is a way to signal to the meta-estimator (like the tablevectorizer) "I'm not the right transformer for this column, pick someone else" .

Ok that makes sense!

Reply:
An error is probably best here?

Reply:
yes an error is probably best. In the TableVectorizer, this situation will not happen because the string preprocessing will have made sure you are dealing with strings, but of course this can be used outside of the tablevectorizer so the check is useful